### DISCUSSION TITLE
How to match an operator?

### DISCUSSION THREAD

**Discussion by pjljvandelaar** [ast-grep/ast-grep#1255] [Q&A]
I would like to find patterns like `$A $BINARY_OPERATOR $A`, but I seem unable to match an operator:
https://ast-grep.github.io/playground.html#eyJtb2RlIjoiUGF0Y2giLCJsYW5nIjoiY3BwIiwicXVlcnkiOiJ4ICRPUEVSQVRPUiB5XHJcbiIsInJld3JpdGUiOiJ5ICRPUEVSQVRPUiB4IiwiY29uZmlnIjoiIiwic291cmNlIjoidm9pZCBtYWluICgpe1xuICBpbnQgeCA9IDM7XG4gIGludCB5ID0gNTtcblxuICBpZiAoeCA8IHkpIHsgYm9vbCBiID0gKHggPT0geSk7IH1cbiJ9

Is matching an operator impossible?

**Comment by HerringtonDarkholme** [comment]
Please read the doc https://ast-grep.github.io/guide/pattern-syntax.html#pattern-matching

<img width="707" alt="image" src="https://github.com/ast-grep/ast-grep/assets/2883231/690dfb9a-d25b-4408-a2e3-d2fe2353d150">

And also the doc https://ast-grep.github.io/guide/rule-config/atomic-rule.html#kind

<img width="730" alt="image" src="https://github.com/ast-grep/ast-grep/assets/2883231/ac6bb3af-f140-4373-9efa-1262718bf966">
